cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -381,12 +376,124 @@ func (m *gceManagerImpl) fetchAutoMigs() error {
 	}
 
 	if changed {
-		return m.cache.RegenerateInstancesCache()
+		return m.RegenerateInstancesCache()
 	}
 
 	return nil
 }
 
+// GetMigForInstance gets mig for given instance ref
+// Attempts to regenerate cache if there is a Mig with matching prefix in migs list.
+// TODO(aleksandra-malinowska): reconsider failing when there's a Mig with
+// matching prefix, but instance doesn't belong to it.
+func (m *gceManagerImpl) GetMigForInstance(instanceRef GceRef) (Mig, error) {
```

Comment:
This was previously done under lock, are we sure we don't need it here?

Reply:
I do not see a reason. gce_manager is not thread safe. And the cache itself is guarded by lock.

Reply:
I believe it's possible to have ongoing regenerateInstancesCache (from goroutine running it every hour, calling regenerateInstanceCacheForMig for every MIG) and regenerateInstanceCacheForMig (from this method) calls. In this case, we need to consider the interactions.

Reply:
Yeah - still I do not believe that it should be a problem. It could have if we depended on updating instances cache at some given time spot (like beginning of the loop). But we did not do that previously - and we do not do that now. 
The worst that can happen if extra `regenerateInstancesCache` is called that we will operate on more "recent" data.